### PR TITLE
KDEV 1626: Add login hooks properties

### PR DESCRIPTION
**Pull request by mkapostolov**
https://kinvey.atlassian.net/browse/KDEV-1626

### CODE REVIEW

File: lib/service/functions.js [hunk 1]
```diff
@@ -40,18 +40,20 @@ function process(task, modules, callback) {
...
   }
 
-  const context = {};
-  const currentContext = task.hookType === 'post' ? task.response : task.request;
+  const { method, username, userId, entityId, loginOptions } = task.request;
+  const context = { method, username, userId, entityId, loginOptions };
+
+  for (const key in context) {
+    if (context[key] == undefined) delete context[key];
+  }
```
Comment (alexanderfilipov) [inline lib/service/functions.js:48]: Well, this looks a bit odd. A simpler logic would be to iterate over a list of keys and assign them to an empty context object if they exist. Something like:
```
const context = {};
for (const key in ['method', 'username', 'userId', 'entityId', 'loginOptions']) {
  if (task.request[key] != null) {
    context[key] = task.request[key];
  }
}
```

File: lib/service/functions.js [hunk 2]
```diff
@@ -40,18 +40,20 @@ function process(task, modules, callback) {
...
   context.headers = currentContext.headers;
-  context.username = task.request.username;
-  context.userId = task.request.userId;
   context.objectName = task.request.objectName || task.request.collectionName;
   context.hookType = task.hookType;
 
-  if (task.request.entityId) {
-    context.entityId = task.request.entityId;
+  if (task.hookType === 'post' && task.response.status > 399) {
+    context.status = task.response.status;
```
Comment (alexanderfilipov) [inline lib/service/functions.js:56]: It seems possible to combine the two checks of `task.hookType === 'post'` into one. Would be simpler IMO.

**Review by alexanderfilipov — CHANGES_REQUESTED** [review]

File: lib/service/functions.js (hunk 2 above)
Comment (mkapostolov) [reply lib/service/functions.js]: Done

File: lib/service/functions.js (hunk 1 above)
Comment (mkapostolov) [reply lib/service/functions.js]: Done

**Comment by mkapostolov** [comment]
Added tests for the changes as well.

**Review by alexanderfilipov — DISMISSED** [review]
LGTM.

**Review by alexanderfilipov — APPROVED** [review]

**Pull request merged**